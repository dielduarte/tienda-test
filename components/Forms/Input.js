import React from 'react';
import { string, func } from 'prop-types';

import {
  Label,
  Input as StyledInput,
  FormRow,
  FormColumn,
  InputPrefix
} from './Ui';

function Input({ label, placeholder, prefix, onChange, errors, ...allProps }) {
  const withPrefix = prefix !== '';
  const withError = errors !== '';

  return (
    <FormRow>
      <FormColumn>
        <Label>{label}</Label>
        <FormRow>
          {withPrefix && <InputPrefix>{prefix}</InputPrefix>}
          <StyledInput
            placeholder={errors || placeholder}
            withPrefix={withPrefix}
            onChange={onChange}
            withError={withError}
            {...allProps}
          />
        </FormRow>
      </FormColumn>
    </FormRow>
  );
}

Input.propsType = {
  label: string.isRequired,
  placeholder: string,
  prefix: string,
  onChange: func,
  errors: string
};

Input.defaultProps = {
  placeholder: '',
  prefix: '',
  errors: '',
  onChange: () => {}
};

export default Input;
