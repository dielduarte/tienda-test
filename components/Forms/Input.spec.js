import { shallow, mount } from 'enzyme';
import Input from './Input';

describe('Input', () => {
  it('should render correcly', () => {
    const wrapper = shallow(
      <Input label="name" name="name" placeholder="your name here" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with prefix', () => {
    const wrapper = shallow(
      <Input label="name" placeholder="your name here" prefix="R$" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with error', () => {
    const wrapper = shallow(
      <Input label="name" prefix="R$" errors="name is required" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange prop when onChange event happen', () => {
    const onChange = jest.fn();

    const wrapper = mount(
      <Input
        label="name"
        placeholder="your name here"
        prefix="R$"
        onChange={onChange}
      />
    );

    wrapper
      .find('input')
      .props()
      .onChange();

    expect(onChange).toHaveBeenCalled();
  });
});
