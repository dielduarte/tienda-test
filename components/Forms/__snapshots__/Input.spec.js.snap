// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correcly 1`] = `
<Styled(styled.div)>
  <Styled(styled.div)>
    <styled.label>
      name
    </styled.label>
    <Styled(styled.div)>
      <styled.input
        name="name"
        onChange={[Function]}
        placeholder="your name here"
        withError={false}
        withPrefix={false}
      />
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Input should render with error 1`] = `
<Styled(styled.div)>
  <Styled(styled.div)>
    <styled.label>
      name
    </styled.label>
    <Styled(styled.div)>
      <styled.div>
        R$
      </styled.div>
      <styled.input
        onChange={[Function]}
        placeholder="name is required"
        withError={true}
        withPrefix={true}
      />
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Input should render with prefix 1`] = `
<Styled(styled.div)>
  <Styled(styled.div)>
    <styled.label>
      name
    </styled.label>
    <Styled(styled.div)>
      <styled.div>
        R$
      </styled.div>
      <styled.input
        onChange={[Function]}
        placeholder="your name here"
        withError={false}
        withPrefix={true}
      />
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;
